.github/workflows/benchmarks.yml

```diff
@@ -0,0 +1,66 @@
+# Run performance benchmarks
+#
+# Continuous benchmarking using pytest-codspeed. Measures the execution speed
+# of tests marked with @pytest.mark.benchmark decorator.
+
+name: Benchmarks
+
+on:
+  # Run on pushes to the main branch
+  push:
+    branches: [ main ]
+  # Uncomment the 'pull_request' line below to trigger the workflow in PR
+  pull_request:
+  # `workflow_dispatch` allows CodSpeed to trigger backtest
+  # performance analysis in order to generate initial data.
+  workflow_dispatch:
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: ${{ github.ref != 'refs/heads/main' }}
+
+jobs:
+  benchmarks:
+    runs-on: ubuntu-22.04
+    defaults:
+      run:
+        shell: bash -l {0}
+
+    steps:
+      # Checkout current git repository
+      - name: Checkout
+        uses: actions/checkout@v4.1.1
+        with:
+          # fetch all history so that setuptools-scm works
+          fetch-depth: 0
+
+      # Install Miniconda with conda-forge dependencies
+      - name: Setup Miniconda
+        uses: conda-incubator/setup-miniconda@v3.0.1
+        with:
+          auto-activate-base: true
+          activate-environment: ""  # base environment
+          channels: conda-forge,nodefaults
+          channel-priority: strict
+
+      # Install GMT and dependencies from conda-forge
+      - name: Install dependencies
+        run: |
+          conda install --solver=libmamba gmt=6.4.0 python=3.12 \
+                        numpy pandas xarray netCDF4 packaging \
+                        pytest pytest-benchmark pytest-mpl
+          $CONDA/bin/python -m pip install -U pytest-codspeed setuptools
```

Comment:
Add `$CONDA/bin` to PATH, so we don't have to use `$CONDA/bin/python`?

Reply:
We need to make sure that conda's python (`/usr/share/miniconda/bin/python`) is used though instead of the system python (`/usr/bin/python`). Is there a way to prepend to $PATH instead of append?

Reply:
https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#adding-a-system-path
prepending is the default.

Reply:
Ah nice, I see this being used at https://github.com/actions/starter-workflows/blob/c31fe3d5d44d7cb4c912f4c3213f7b4610f13ea2/ci/python-package-conda.yml#L19-L20 too, but assumed it was appending. Done at f45b727ce0c389d022398314a84464b4977cb330, but I might make another commit to use the `make ...` commands instead of `python -m ...`.